aiida/orm/utils/mixins.py

```diff
@@ -105,13 +109,47 @@ def _set_function_starting_line_number(self, function_starting_line_number):
         """
         self.set_attribute(self.FUNCTION_STARTING_LINE_KEY, function_starting_line_number)
 
+    @property
+    def function_number_of_lines(self):
+        """Return the number of lines of the wrapped function in its source file.
+
+        :returns: the number of lines or None
+        """
+        return self.get_attribute(self.FUNCTION_NUMBER_OF_LINES_KEY, -1)
+
+    def _set_function_number_of_lines(self, function_number_of_lines):
+        """Set the number of lines of the wrapped function in its source file.
+
+        :param function_number_of_lines: the number of lines
+        """
+        self.set_attribute(self.FUNCTION_NUMBER_OF_LINES_KEY, function_number_of_lines)
+
     def get_function_source_code(self):
-        """Return the absolute path to the source file in the repository.
+        """Return the content of the entire source file that contains the process function.
 
-        :returns: the absolute path of the source file in the repository, or None if it does not exist
+        :returns: the content of the source file
         """
+        warnings.warn(
+            'in the future this will return only the source of the function and'
+            ' please use `get_function_source_file` to get'
+            ' the entire content of the file instead', AiidaDeprecationWarning
+        )
```

Comment:
So, there are a couple of comments regarding this depracation you are doing:

1. You need to already include the `get_function_source_file` method so that users can immediately start fixing their scripts with the new way of doing it.
2. The depracation path for this is a bit complicated. I know you discussed with @sphuber in the general thread what would be the *hypothetical* way to do this, but I'm not sure he was actually agreeing it would be a good idea to go this way (maybe he can confirm in case I misinterpreted?). I personally agree that it is better "namewise", but changing to that might generate a bit of annoyance on the users.
3. The warning itself it is a bit unclear to me, I would maybe add a legthier description if we are going to go this way. Something like:

```suggestion
        warnings.warn(
            'you are currently using `get_function_source_code` to get the entire'
            ' file where the process is defined, but in the future this method will'
            ' be used to return only the part of that file that corresponds to the'
            ' process itself. Please use the method `get_function_source_file`'
            ' instead.', AiidaDeprecationWarning
        )
```

Reply:
> You need to already include the get_function_source_file method so that users can immediately start fixing their scripts with the new way of doing it.

My bad. add it as the implemented function and forward it to `get_function_source_code`. 

> The depracation path for this is a bit complicated. I know you discussed with @sphuber in the general thread what would be the hypothetical way to do this, but I'm not sure he was actually agreeing it would be a good idea to go this way (maybe he can confirm in case I misinterpreted?). I personally agree that it is better "namewise", but changing to that might generate a bit of annoyance on the users.

However, I can not figure out a better way to do this. I think it can reduce the burden to users only by a clear docstring and comprehensive warning information.

> The warning itself it is a bit unclear to me, I would maybe add a legthier description if we are going to go this way. 

Thanks! changed as you recommend.

Reply:
> However, I can not figure out a better way to do this

So, I think the alternative is to accept that `get_function_source_code` now returns the full file and create a new name for the one that just returns the function (`get_function_source_def`?). This is less consistent with the original information (where `source_file` had the file and `source_code` had the definition) but more consistent with the current status (where `source_code` is the name of the file) and also causes less friction with the users because there is no deprecations needed. This was what I understood from the first comment that @sphuber made:

> You should revert the change of get_function_source_code and keep it as it was, i.e. return the entire file. You can then add a method that returns just the source code of the function, but you will have to find a different method name for this.

There was no mention of any further change there, just to rename the new function and leave the old one as it was. Then the longer path in the latter comment read more like a hipothetical situation than a prescription for this case.

Again, I am personally in favour of doing the longer deprecation path, I just want to make sure that we are all on the same page and there is not misunderstandings or something we are not considering.

Reply:
@ramirezfranciscof  Thanks for the clarification. So in this simple pr, only need to add a new method for section of source code, then left deprecation and renaming to the future, correct? I rebase the PR and please have a look at your available.